<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1116</width>
    <height>629</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>1000</width>
    <height>600</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true">QMainWindow {
	overflow: scroll;
	background-color: #343436;
}

QGroupBox {
	background-color: #393939;
	border-radius: 8px;
	margin: 8px;
	font-size: 20px;
	color: white;
}

QGroupBox * {
	color: white;
}

QGroupBox QLabel {
	font-size: 18px;
}

QComboBox, QComboBox QAbstractItemView {
	background-color: #303032;
}

QSpinBox, QDoubleSpinBox {
	background-color: #343436;
}

.group-header {
	font-size: 21px;
	font-weight: bold;
}
</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_13">
    <item row="0" column="0">
     <widget class="QWidget" name="widget" native="true">
      <property name="styleSheet">
       <string notr="true">QWidget {
	background-color: #343436;
}

QGroupBox {
	background-color: #393939;
	border-radius: 8px;
	margin: 8px;
	font-size: 20px;
	color: white;
}

QGroupBox * {
	color: white;
	background-color: #393939;
}</string>
      </property>
      <layout class="QGridLayout" name="gridLayout">
       <item row="0" column="0">
        <widget class="QScrollArea" name="scrollArea">
         <property name="styleSheet">
          <string notr="true"/>
         </property>
         <property name="widgetResizable">
          <bool>true</bool>
         </property>
         <widget class="QWidget" name="scrollAreaWidgetContents">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>-288</y>
            <width>512</width>
            <height>879</height>
           </rect>
          </property>
          <layout class="QGridLayout" name="gridLayout_7">
           <item row="0" column="0">
            <widget class="QGroupBox" name="segmenation_section">
             <property name="styleSheet">
              <string notr="true"/>
             </property>
             <property name="title">
              <string/>
             </property>
             <property name="alignment">
              <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
             </property>
             <layout class="QGridLayout" name="gridLayout_2">
              <item row="1" column="0">
               <widget class="QSpinBox" name="segmentation_input">
                <property name="styleSheet">
                 <string notr="true">margin-top: 10px
</string>
                </property>
                <property name="maximum">
                 <number>1000</number>
                </property>
                <property name="value">
                 <number>127</number>
                </property>
               </widget>
              </item>
              <item row="0" column="0">
               <widget class="QLabel" name="label_7">
                <property name="text">
                 <string>Segmentation</string>
                </property>
                <property name="class" stdset="0">
                 <string>group-header</string>
                </property>
               </widget>
              </item>
              <item row="1" column="1">
               <widget class="QPushButton" name="segmentation_ok">
                <property name="cursor">
                 <cursorShape>PointingHandCursor</cursorShape>
                </property>
                <property name="styleSheet">
                 <string notr="true">QPushButton {
	background-color: #616264;
	font-size: 18px;
}</string>
                </property>
                <property name="text">
                 <string>Apply</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QGroupBox" name="smoothing_section">
             <property name="title">
              <string/>
             </property>
             <layout class="QGridLayout" name="gridLayout_4">
              <item row="1" column="0" colspan="2">
               <widget class="QLabel" name="label_2">
                <property name="text">
                 <string>Kernel Size:</string>
                </property>
               </widget>
              </item>
              <item row="2" column="1" colspan="2">
               <widget class="QComboBox" name="linear_input">
                <item>
                 <property name="text">
                  <string>Box Filter</string>
                 </property>
                </item>
               </widget>
              </item>
              <item row="2" column="3">
               <widget class="QPushButton" name="smoothing_linear_ok">
                <property name="cursor">
                 <cursorShape>PointingHandCursor</cursorShape>
                </property>
                <property name="styleSheet">
                 <string notr="true">QPushButton {
	background-color: #616264;
	font-size: 18px;
}</string>
                </property>
                <property name="text">
                 <string>Apply</string>
                </property>
               </widget>
              </item>
              <item row="0" column="0" colspan="3">
               <widget class="QLabel" name="label_11">
                <property name="text">
                 <string>Smoothing Filters</string>
                </property>
                <property name="class" stdset="0">
                 <string>group-header</string>
                </property>
               </widget>
              </item>
              <item row="2" column="0">
               <widget class="QLabel" name="label_3">
                <property name="text">
                 <string>Linear:</string>
                </property>
               </widget>
              </item>
              <item row="3" column="3">
               <widget class="QPushButton" name="smoothing_nonlinear_ok">
                <property name="cursor">
                 <cursorShape>PointingHandCursor</cursorShape>
                </property>
                <property name="styleSheet">
                 <string notr="true">QPushButton {
	background-color: #616264;
	font-size: 18px;
}</string>
                </property>
                <property name="text">
                 <string>Apply</string>
                </property>
               </widget>
              </item>
              <item row="3" column="1" colspan="2">
               <widget class="QComboBox" name="nonlinear_input">
                <item>
                 <property name="text">
                  <string>Min Filter</string>
                 </property>
                </item>
               </widget>
              </item>
              <item row="1" column="2">
               <widget class="QComboBox" name="kernel_size_input">
                <property name="editable">
                 <bool>false</bool>
                </property>
                <property name="currentText">
                 <string>3 x 3</string>
                </property>
                <property name="currentIndex">
                 <number>0</number>
                </property>
                <item>
                 <property name="text">
                  <string>3 x 3</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>5 x 5</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>7 x 7</string>
                 </property>
                </item>
               </widget>
              </item>
              <item row="3" column="0">
               <widget class="QLabel" name="label_4">
                <property name="text">
                 <string>Non-Linear:</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item row="3" column="0">
            <widget class="QGroupBox" name="sharpening_section">
             <property name="title">
              <string/>
             </property>
             <layout class="QGridLayout" name="gridLayout_5">
              <item row="2" column="0">
               <widget class="QLabel" name="label_6">
                <property name="text">
                 <string>High Boost:</string>
                </property>
               </widget>
              </item>
              <item row="2" column="1">
               <widget class="QDoubleSpinBox" name="high_boost_inout">
                <property name="styleSheet">
                 <string notr="true"/>
                </property>
               </widget>
              </item>
              <item row="0" column="0" colspan="2">
               <widget class="QLabel" name="label_13">
                <property name="text">
                 <string>Sharpening Filters</string>
                </property>
                <property name="class" stdset="0">
                 <string>group-header</string>
                </property>
               </widget>
              </item>
              <item row="1" column="1">
               <widget class="QCheckBox" name="laplacian_enhaced_check">
                <property name="cursor">
                 <cursorShape>PointingHandCursor</cursorShape>
                </property>
                <property name="text">
                 <string>Enhanced</string>
                </property>
               </widget>
              </item>
              <item row="1" column="0">
               <widget class="QLabel" name="label_5">
                <property name="text">
                 <string>Laplacian:</string>
                </property>
               </widget>
              </item>
              <item row="3" column="2">
               <widget class="QPushButton" name="sharpening_ok">
                <property name="cursor">
                 <cursorShape>PointingHandCursor</cursorShape>
                </property>
                <property name="styleSheet">
                 <string notr="true">QPushButton {
	background-color: #616264;
	font-size: 18px;
}</string>
                </property>
                <property name="text">
                 <string>Apply</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item row="4" column="0">
            <widget class="QGroupBox" name="detection_section">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="title">
              <string/>
             </property>
             <layout class="QGridLayout" name="gridLayout_6">
              <item row="1" column="0">
               <widget class="QLabel" name="label_9">
                <property name="text">
                 <string>Filter:</string>
                </property>
               </widget>
              </item>
              <item row="0" column="0" colspan="3">
               <widget class="QLabel" name="label_14">
                <property name="text">
                 <string>Edge Detection</string>
                </property>
                <property name="class" stdset="0">
                 <string>group-header</string>
                </property>
               </widget>
              </item>
              <item row="1" column="1" colspan="2">
               <widget class="QComboBox" name="detection_filter_input">
                <item>
                 <property name="text">
                  <string>Sobel Filter</string>
                 </property>
                </item>
               </widget>
              </item>
              <item row="2" column="0">
               <widget class="QLabel" name="label_10">
                <property name="text">
                 <string>Type:</string>
                </property>
               </widget>
              </item>
              <item row="2" column="1" colspan="2">
               <widget class="QComboBox" name="nonlinear_input_2">
                <item>
                 <property name="text">
                  <string>Vertical</string>
                 </property>
                </item>
               </widget>
              </item>
              <item row="3" column="2">
               <widget class="QPushButton" name="detection_ok">
                <property name="cursor">
                 <cursorShape>PointingHandCursor</cursorShape>
                </property>
                <property name="styleSheet">
                 <string notr="true">QPushButton {
	background-color: #616264;
	font-size: 18px;
}</string>
                </property>
                <property name="text">
                 <string>Apply</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QGroupBox" name="contrast_section">
             <property name="title">
              <string/>
             </property>
             <layout class="QGridLayout" name="gridLayout_3">
              <item row="0" column="0">
               <widget class="QLabel" name="label_12">
                <property name="text">
                 <string>Contrast Stretching</string>
                </property>
                <property name="class" stdset="0">
                 <string>group-header</string>
                </property>
               </widget>
              </item>
              <item row="1" column="0">
               <widget class="QLabel" name="label">
                <property name="text">
                 <string>Histogram Equalization</string>
                </property>
               </widget>
              </item>
              <item row="1" column="1">
               <widget class="QPushButton" name="histogram_ok">
                <property name="cursor">
                 <cursorShape>PointingHandCursor</cursorShape>
                </property>
                <property name="styleSheet">
                 <string notr="true">QPushButton {
	background-color: #616264;
	font-size: 18px;
}</string>
                </property>
                <property name="text">
                 <string>Apply</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item row="5" column="0">
            <widget class="QGroupBox" name="transmission_section">
             <property name="title">
              <string/>
             </property>
             <layout class="QGridLayout" name="gridLayout_8">
              <item row="3" column="0">
               <widget class="QLabel" name="label_17">
                <property name="text">
                 <string>Time:</string>
                </property>
               </widget>
              </item>
              <item row="2" column="0">
               <widget class="QLabel" name="label_15">
                <property name="text">
                 <string>Baudrate:</string>
                </property>
               </widget>
              </item>
              <item row="0" column="0" colspan="3">
               <widget class="QLabel" name="label_16">
                <property name="text">
                 <string>Transmission Time</string>
                </property>
                <property name="class" stdset="0">
                 <string>group-header</string>
                </property>
               </widget>
              </item>
              <item row="2" column="1">
               <widget class="QSpinBox" name="baud_input">
                <property name="maximum">
                 <number>999999999</number>
                </property>
                <property name="stepType">
                 <enum>QAbstractSpinBox::DefaultStepType</enum>
                </property>
               </widget>
              </item>
              <item row="2" column="2">
               <widget class="QLabel" name="label_8">
                <property name="text">
                 <string>K (x1000)</string>
                </property>
               </widget>
              </item>
              <item row="3" column="2">
               <widget class="QLabel" name="label_18">
                <property name="text">
                 <string>seconds</string>
                </property>
               </widget>
              </item>
              <item row="3" column="1">
               <widget class="QLCDNumber" name="transmission_time_output">
                <property name="smallDecimalPoint">
                 <bool>false</bool>
                </property>
                <property name="digitCount">
                 <number>19</number>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QWidget" name="widget_2" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <layout class="QGridLayout" name="gridLayout_11">
       <item row="0" column="0">
        <widget class="QGroupBox" name="groupBox">
         <property name="styleSheet">
          <string notr="true"/>
         </property>
         <property name="title">
          <string/>
         </property>
         <property name="alignment">
          <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
         </property>
         <layout class="QGridLayout" name="gridLayout_10">
          <item row="1" column="0">
           <widget class="QPushButton" name="import_image">
            <property name="cursor">
             <cursorShape>PointingHandCursor</cursorShape>
            </property>
            <property name="styleSheet">
             <string notr="true">QPushButton {
	background-color: #616264;
	font-size: 18px;
}</string>
            </property>
            <property name="text">
             <string>Import</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QWidget" name="widget_3" native="true">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <layout class="QGridLayout" name="gridLayout_14">
             <item row="0" column="0">
              <widget class="QLabel" name="image_in">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="maximumSize">
                <size>
                 <width>290</width>
                 <height>275</height>
                </size>
               </property>
               <property name="styleSheet">
                <string notr="true">margin-right: 10px;</string>
               </property>
               <property name="text">
                <string/>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QLabel" name="image_out">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="minimumSize">
                <size>
                 <width>290</width>
                 <height>275</height>
                </size>
               </property>
               <property name="styleSheet">
                <string notr="true">margin-left: 10px;</string>
               </property>
               <property name="text">
                <string/>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QGroupBox" name="groupBox_2">
         <property name="title">
          <string/>
         </property>
         <layout class="QGridLayout" name="gridLayout_12">
          <item row="0" column="0">
           <widget class="QLabel" name="label_22">
            <property name="text">
             <string>Histogram</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="histogram_in">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="maximumSize">
             <size>
              <width>494</width>
              <height>228</height>
             </size>
            </property>
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
